frontend/packages/console-shared/src/components/dashboard/utilization-card/prometheus-hook.ts

```diff
@@ -35,4 +38,41 @@ export const usePrometheusQuery: UsePrometheusQuery = (query, humanize) => {
   return results;
 };
 
+const customSelectorCreator = createSelectorCreator(defaultMemoize, shallowEqual);
+
+export const usePrometheusQueries: UsePrometheusQueries = (queries, metric) => {
+  const dispatch = useDispatch();
+  React.useEffect(() => {
+    queries.forEach((query) => dispatch(watchPrometheusQuery(query)));
+    return () => {
+      queries.forEach((query) => dispatch(stopWatchPrometheusQuery(query)));
+    };
+  }, [dispatch, queries]);
+
+  const selectors = React.useMemo(
+    () => queries.map((q) => ({ dashboards }) => dashboards.getIn([RESULTS_TYPE.PROMETHEUS, q])),
+    [queries],
+  );
+
+  const querySelector = React.useMemo(() => customSelectorCreator(selectors, (...data) => data), [
+    selectors,
+  ]);
+
+  const queryResults = useSelector<RootState, ImmutableMap<string, any>>(querySelector);
+
+  const results = React.useMemo<UsePrometheusQueriesResult>(() => {
+    if (_.isEmpty(queryResults?.[0])) {
+      return queries.map(() => [null, true, null]) as UsePrometheusQueriesResult;
+    }
+    return queryResults.reduce((acc, curr) => {
+      const value = getInstantVectorStats(curr.get('data'), metric);
+      return [...acc, [value, false, curr.get('loadError')]];
```

Comment:
you need to check if the data are set, if not then the query is still pending
```suggestion
      return [...acc, [value, !curr.get('data'), curr.get('loadError')]];
```